conda_build/post.py

```diff
@@ -848,6 +849,44 @@ def library_nature(
 ]
 
 
+def _expand_dsolist(dsolist):
+    result = []
+    for d in dsolist:
+        if d.startswith("*"):
+            result.append(d)
+        else:
+            result.extend([f"**/{basename(f)}" for f in utils.glob(d)])
+    return result
+
+
+def _get_dsolists_prefix(prefix, subdir):
+    allow = []
+    deny = []
+    for fname in utils.glob(join(prefix, "etc", "conda-build", "dsolists.d", "*.json")):
```

Comment:
```suggestion
    for fname in sorted(utils.glob(join(prefix, "etc", "conda-build", "dsolists.d", "*.json"), recursive=True, include_hidden=True)):
```

I think you want to recurse and add hidden paths if matching. Also, let's sort the result for some more reproducibility in case of errors.